### PR TITLE
addTrack: split up tests and reduce dependencies

### PR DESCRIPTION
**Pull request by fippo**
splits up addTrack tests and reduce dependencies. Makes more tests pass in Chrome since it removes the dependency on getTransceivers

Discussion item for https://www.w3.org/2011/04/webrtc/wiki/April_26_2018 next Thursday

### CODE REVIEW

File: webrtc/RTCPeerConnection-addTrack.https.html [hunk 1]
```diff
@@ -64,6 +64,62 @@
...
+  promise_test(t => {
+    return addTrackFromGetUserMedia(t)
+      .then(({pc, sender}) => {
+        const transceivers = pc.getTransceivers();
+        assert_equals(transceivers.length, 1,
+          'Expect only one transceiver');
+        const [transceiver] = transceivers;
+        assert_equals(transceiver.sender, sender, 'transceivers sender equals sender');
+      });
+  }, 'addTrack adds an RTCRtpTransceiver whose sender is set to the sender returned by addTrack');
```
Comment (fippo) [inline webrtc/RTCPeerConnection-addTrack.https.html]: nit: creates an RTCRtpTransceiver for consistency

File: webrtc/RTCPeerConnection-addTrack.https.html [hunk 2]
```diff
@@ -64,6 +64,62 @@
               transceiver be the result.
           4.  Add transceiver to connection's set of transceivers.
    */
+  function addTrackFromGetUserMedia(t) {
```
Comment (fippo) [inline webrtc/RTCPeerConnection-addTrack.https.html]: might be better to call `setup()`, `setupTest()` etc

Comment (henbos) [reply webrtc/RTCPeerConnection-addTrack.https.html]: And to use async function and await getUserMedia?

File: webrtc/RTCPeerConnection-addTrack.https.html [hunk 3]
```diff
@@ -64,6 +64,62 @@
...
+    return navigator.mediaDevices.getUserMedia({ audio: true })
+    .then(mediaStream => {
+      const track = mediaStream.getTracks()[0];
+      t.add_cleanup(() => track.stop());
+      const sender = pc.addTrack(track, mediaStream);
+      return {pc, track, sender};
+    });
+  }
+
+  promise_test(t => {
```
Comment (henbos) [inline webrtc/RTCPeerConnection-addTrack.https.html]: Here and elsewhere: Can you update to use promise_test(async t => { ... while you're at it? :) at least for the tests you're updating.

File: webrtc/RTCPeerConnection-addTrack.https.html [hunk 4]
```diff
@@ -81,21 +137,6 @@
...
-
-      assert_array_equals([sender], pc.getSenders(),
-        'Expect only one sender with given track added');
-
-      const { receiver } = transceiver;
-      assert_equals(receiver.track.kind, 'audio');
-      assert_array_equals([transceiver.receiver], pc.getReceivers(),
-        'Expect only one receiver associated with transceiver added');
     });
   }, 'addTrack with single track argument and no mediaStream should succeed');
```
Comment (henbos) [inline webrtc/RTCPeerConnection-addTrack.https.html]: Optional: There are other tests that could use the same setup function as above, if you care to drive-by.

**Review by henbos — COMMENTED** [review]
I like your improvements, thanks! My only comments are a wish for switching to async while we're at it.

File: webrtc/RTCPeerConnection-addTrack.https.html (hunk 2 above)
Comment (fippo) [reply webrtc/RTCPeerConnection-addTrack.https.html]: done locally.

File: webrtc/RTCPeerConnection-addTrack.https.html (hunk 3 above)
Comment (fippo) [reply webrtc/RTCPeerConnection-addTrack.https.html]: done locally.

File: webrtc/RTCPeerConnection-addTrack.https.html (hunk 4 above)
Comment (fippo) [reply webrtc/RTCPeerConnection-addTrack.https.html]: i know. this is just a strawman thing :-)

**Comment by fippo** [comment]
we went another route.
